packages/back-end/src/models/EventWebhookModel.ts

```diff
@@ -264,16 +368,38 @@ export const getAllEventWebHooks = async (
  * @param eventName
  * @param enabled
  */
-export const getAllEventWebHooksForEvent = async (
-  organizationId: string,
-  eventName: NotificationEventName,
-  enabled: boolean
-): Promise<EventWebHookInterface[]> => {
-  const docs = await EventWebHookModel.find({
+export const getAllEventWebHooksForEvent = async ({
+  organizationId,
+  eventName,
+  enabled,
+  tags,
+  projects,
+}: {
+  organizationId: string;
+  eventName: NotificationEventName;
+  enabled: boolean;
+  tags: string[];
+  projects: string[];
+}): Promise<EventWebHookInterface[]> => {
+  const allDocs = await EventWebHookModel.find({
     organizationId,
     events: eventName,
     enabled,
   });
 
+  const docs = allDocs.filter((doc) => {
+    if (doc.tags?.length && tags?.length && !intersection(doc.tags, tags))
+      return false;
+
+    if (
```

Comment:
possibly yes, but not 100% sure if this captures the desired behavior. I *think* this is what we're looking for; we should double check these scenarios:
- doc has no projects, event has no projects: allow
- doc has projects, event has no projects: allow
- doc has no projects, event has projects: remove
- doc has projects, event has projects: only allow if intersecting

Reply:
I added a test for that: https://github.com/growthbook/growthbook/pull/2150/files#diff-b6d0babef2699f106a9f4ddb27df1aaee061c3090d237cf6a13b96e9a8d09d9d